src/ui/theme.cpp

```diff
@@ -592,16 +596,18 @@ void Theme::measureLayer(const Widget* widget,
     case Style::Layer::Type::kText:
       if (layer.color() != gfx::ColorNone) {
         gfx::Size textSize;
-        if (style->font() &&
-            style->font() != widget->font()) {
-          text::Font* font = style->font();
-          textSize = gfx::Size(Graphics::measureUITextLength(widget->text(), font),
-                               font->height());
-        }
-        else {
-          // We can use Widget::textSize() because we're going to use
-          // the widget font and, probably, the cached TextBlob.
-          textSize = widget->textSize();
+        if (!widget->text().empty()) {
```

Comment:
It looks like we cannot filter out this case because sometimes we "size hint" widgets with empty text, e.g. with this change:

![Screenshot from 2024-08-22 20-15-45](https://github.com/user-attachments/assets/cdcea9e4-13c8-438c-83ce-fed33c15238b)

![Screenshot from 2024-08-22 20-15-51](https://github.com/user-attachments/assets/fb855a84-b073-49b1-8227-71ab54e78251)

While just commenting this condition we get the expected result:

![Screenshot from 2024-08-22 20-16-37](https://github.com/user-attachments/assets/e1725b94-8a7b-4e34-be8d-04c36efd9c84)

![Screenshot from 2024-08-22 20-16-31](https://github.com/user-attachments/assets/d0336b27-b843-411a-9704-22c273a2008c)

Reply:
Should still avoid drawing it and use the font height or similar value to get the size right, I can do the change if you want

Reply:
Don't worry, I'll take from here testing in all platforms and making comments just to leave a record about it.